File: leetcode/242_valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s)!=len(t):
            return False
        freq1 = dict()
        freq2 = dict()
        for ch in s:
            freq1[ch] = freq1.get(ch, 0) + 1
        for ch in t:
            freq2[ch] = freq2.get(ch, 0) + 1
        for ch in s:
            if freq1[ch] != freq2.get(ch, 0):
                return False
        return True

File: leetcode/242_valid_anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length()!=t.length())
            return false;
        int[] count = new int[26];
        for(int i=0; i<s.length(); i++){
            int pos = s.charAt(i) - 'a';
            count[pos] += 1;
        }
        for(int i=0; i<t.length(); i++){
            int pos = t.charAt(i) - 'a';
            count[pos] -= 1;
            if (count[pos] < 0)
                return false;
        }
        return true;
    }
}
